aocd/runner.py

```diff
@@ -320,6 +353,11 @@ def run_for(
             continue
         entry_point = eps[plugin]
         puzzle = Puzzle(year, day)
+        # TODO: this Union is awkward (b/c it means we need the
+        # `assert isinstance` checks later on.)
+        # There's probably a way to clean this up, but it likely requires a
+        # bit of refactoring
+        datas: Union[range, Mapping[str, str]]
```

Comment:
Yeah it would be good to refactor it so that the Union isn't needed. I don't love the way this was done originally where "datas" is some totally different structure depending on whether you're running against real data or example data, it's sort of wacky.

Reply:
Looks like you could just use `datas = [ex.input_data for ex in Puzzle(year, day).examples]`.